sources/Valkey.Glide/BaseClient.ScriptingCommands.cs

```diff
@@ -0,0 +1,378 @@
+    }
+
+    /// <inheritdoc/>
+    public async Task<ValkeyResult> FCallAsync(
+        string function,
+        string[] keys,
+        string[] args,
+        CommandFlags flags = CommandFlags.None,
+        CancellationToken cancellationToken = default)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+        return await Command(Request.FCallAsync(function, keys, args));
+    }
+
+    /// <inheritdoc/>
+    public async Task<ValkeyResult> FCallReadOnlyAsync(
+        string function,
+        CommandFlags flags = CommandFlags.None,
+        CancellationToken cancellationToken = default)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+        return await Command(Request.FCallReadOnlyAsync(function, null, null));
+    }
+
+    /// <inheritdoc/>
+    public async Task<ValkeyResult> FCallReadOnlyAsync(
+        string function,
+        string[] keys,
+        string[] args,
+        CommandFlags flags = CommandFlags.None,
+        CancellationToken cancellationToken = default)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+        return await Command(Request.FCallReadOnlyAsync(function, keys, args));
+    }
+
+    // ===== Function Management =====
+
+    /// <inheritdoc/>
+    public async Task<string> FunctionLoadAsync(
+        string libraryCode,
+        bool replace = false,
+        CommandFlags flags = CommandFlags.None,
+        CancellationToken cancellationToken = default)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+        return await Command(Request.FunctionLoadAsync(libraryCode, replace));
+    }
+
+    /// <inheritdoc/>
+    public async Task<string> FunctionFlushAsync(
+        CommandFlags flags = CommandFlags.None,
+        CancellationToken cancellationToken = default)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+        return await Command(Request.FunctionFlushAsync());
+    }
+
+    /// <inheritdoc/>
+    public async Task<string> FunctionFlushAsync(
+        FlushMode mode,
+        CommandFlags flags = CommandFlags.None,
+        CancellationToken cancellationToken = default)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+        return await Command(Request.FunctionFlushAsync(mode));
+    }
+
+    // ===== StackExchange.Redis Compatibility Methods =====
+
+    /// <inheritdoc/>
+    public async Task<ValkeyResult> ScriptEvaluateAsync(string script, ValkeyKey[]? keys = null, ValkeyValue[]? values = null,
+        CommandFlags flags = CommandFlags.None)
+    {
+        if (string.IsNullOrEmpty(script))
```

Comment:
We are checking for null script here in `ScriptEvaluateAsync`, but we are not doing that in `InvokeScriptAsync` in this class or in `GlideClusterClient.ScriptingCommands` (could apply to some other parameters as well, it is inconsistent whether we check for null/empty or not).